### ISSUE TITLE
When pod opens port, container cannot use it

### ISSUE THREAD

**Issue by Yajo** [containers/libpod#6600]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

When you define open ports in the pod, the container cannot listen to them!

**Steps to reproduce the issue:**

1. `sudo podman pod create --name test --publish 80:80`
2. `sudo podman container run --rm --pod test -it traefik` (this image listens on port 80 by default)

**Describe the results you received:**

```bash
➤ sudo podman pod create --name test --publish 80:80
7f04e8d1fa7b01ececc8652468f966224a0d7f461c55b9434dea98aeb7c993e4

➤ sudo podman container run --rm --pod test -it traefik
Trying to pull registry.fedoraproject.org/traefik...
  manifest unknown: manifest unknown
Trying to pull registry.access.redhat.com/traefik...
  name unknown: Repo not found
Trying to pull registry.centos.org/traefik...
  manifest unknown: manifest unknown
Trying to pull docker.io/library/traefik...
Getting image source signatures
Copying blob 2be216d464d0 done  
Copying blob f16506d32a25 done  
Copying blob cbdbe7a5bc2a done  
Copying blob aed51f08b610 done  
Copying config 3d0acfd4b5 done  
Writing manifest to image destination
Storing signatures
ERRO[0007] error starting some container dependencies   
ERRO[0007] "cannot listen on the TCP port: listen tcp4 :80: bind: address already in use" 
Error: error starting some containers: internal libpod error
```

**Describe the results you expected:**
I expect that if I declare in the pod that I'm going to use some container's port 80, then later when a container listens there, it should be able to. Otherwise, how am I supposed to use the port? :shrug: 

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
➤ podman version
Version:            1.9.3
RemoteAPI Version:  1
Go Version:         go1.14.2
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**

```
➤ sudo podman info --debug
debug:
  compiler: gc
  gitCommit: ""
  goVersion: go1.14.2
  podmanVersion: 1.9.3
host:
  arch: amd64
  buildahVersion: 1.14.9
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.17-1.fc32.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.17, commit: bb8e273f5925c1a51737644637ef65d094a67ab1'
  cpus: 12
  distribution:
    distribution: fedora
    version: "32"
  eventLogger: file
  hostname: yajolap-tecnativa-com
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.6.16-300.fc32.x86_64
  memFree: 2161668096
  memTotal: 16616148992
  ociRuntime:
    name: crun
    package: crun-0.13-2.fc32.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.13
      commit: e79e4de4ac16da0ce48777afb72c6241de870525
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  rootless: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 8376020992
  swapTotal: 8376020992
  uptime: 1h 38m 35.22s (Approximately 0.04 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageStore:
    number: 0
  runRoot: /var/run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
➤ rpm -q podman
podman-1.9.3-1.fc32.x86_64
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Physical Fedora Silverblue 32.

**Comment by zhangguanzhang** [comment]
I have run the following without any problem. Are you sure that there is no process bind the port 80 on the system before executing the first command?

**Comment by mheon** [comment]
The key line here is `ERRO[0007] error starting some container dependencies   ` which tells us this is not an error starting the container in question, but actually a container it depends on - in this case, the infra container of the pod. I concur with @zhangguanzhang that the most likely cause is something on your system is already bound to port 80.

The error message there could probably do with some improvement, though

**Comment by Yajo** [comment]
Oh I think I found the problem. I had another empty pod with the same ports. Try this:

```bash
sudo podman pod create --name test1 --publish 80:80
sudo podman pod create --name test2 --publish 80:80
sudo podman container run --rm --pod test2 -it traefik --log.level debug
```

Creating pod test2 doesn't fail, so when booting the container, the head-scratch is mandatory. I guess there should be a warning when starting that pod.

**Comment by zhangguanzhang** [comment]
The Common process forwards the bind port to the container only when the container is running

**Comment by mheon** [comment]
It's perfectly legal to have as many containers bind to port 80 as you want - but only 1 can actually be running at a time, because the port will be in use.

**Comment by Yajo** [comment]
Yikes, now it's working too.

This is frustrating, I had nothing listening on port 80 when tested.

I'm on a different machine now. Is it maybe because I switched the firewalld backend to iptables in the other machine where that was failing? :thinking:

**Comment by mheon** [comment]
Probably not? This specific errors comes from a different part of the code from firewall config - to ensure that other programs on the host don't use a port while it's forwarded to the container we will bind the port ourselves (firewall config means traffic will still go to the container even if that happens, but people get very confused as to why the web server successfully bound to port 80 isn't getting traffic, forgetting they also have a container running on 80). The error is coming from there - Podman was unable to bind to 80 in our attempt to block other programs on the host from using the port.

**Comment by Yajo** [comment]
Today I tried on that other machine and it went well also.

Definitely I must have had something listening in port 80. I can't believe I went to open the issue before checking that... :facepalm: 

Thanks and sorry!

**Comment by stellarpower** [comment]
I've hit a similar issue; admittedly on an old release (stuck with Bionic)

I have a pod with a reverse proxy and some app containers. I'll call it old-pod. I needed new ports open so I had to change the setup and make a new pod, new-pod. 

So old-pod and new-pod both listen on 8080 and 8443 for the reverse proxy. The below commands are simulated and slightly arbitrary, as I'm using ansible, but:

```
$ podman pod stop old-pod
$ podman pod ps -a:
old-pod .... 0.0.0.0:8080->80 0.0.0.0.8443->443 
new-pod .... 0.0.0.0:8080->80 0.0.0.0.8443->443 0.0.0.0:1000->1000
$ podman start new-pod
 .... Error: internal libpod error - 127.0.0.1:localhost address already in use
 ```
 
 However, once I destroyed the old pod before starting the new, everything ran smoothly.
